src/tests/Microsoft.PowerFx.Interpreter.Tests.Shared/RecalcEngineTests.cs

```diff
@@ -679,33 +681,56 @@ public void ImperativeUserDefinedFunctionTest(string udfExpression, string expre
         [Fact]
 
         public void DelegatableUDFTest()
-        {
-            var config = new PowerFxConfig();
-            config.EnableSetFunction();
-
+        {
+            var symbolTable = new DelegatableSymbolTable();
             var schema = DType.CreateTable(
                 new TypedName(DType.Guid, new DName("ID")),
                 new TypedName(DType.Number, new DName("Value")));
-            config.SymbolTable.AddEntity(new TestDataSource("MyDataSource", schema));
-            config.SymbolTable.AddType(new DName("MyDataSourceTableType"), FormulaType.Build(schema));
+            symbolTable.AddEntity(new TestDelegableDataSource(
+                "MyDataSource",
+                schema,
+                new TestDelegationMetadata(
+                        new DelegationCapability(DelegationCapability.Filter),
+                        schema,
+                        new FilterOpMetadata(
+                            schema,
+                            new Dictionary<DPath, DelegationCapability>(),
+                            new Dictionary<DPath, DelegationCapability>(),
+                            new DelegationCapability(DelegationCapability.GreaterThan),
+                            null)),
+                true));
+            symbolTable.AddType(new DName("MyDataSourceTableType"), FormulaType.Build(schema));
+            var config = new PowerFxConfig()
+            {
+                SymbolTable = symbolTable
+            };
+            config.EnableSetFunction();
 
             var recalcEngine = new RecalcEngine(config);
 
-            recalcEngine.AddUserDefinedFunction("A():MyDataSourceTableType = Filter(Sort(MyDataSource,Value), Value > 10);C():MyDataSourceTableType = A();", CultureInfo.InvariantCulture, symbolTable: recalcEngine.EngineSymbols, allowSideEffects: true);
+            recalcEngine.AddUserDefinedFunction("A():MyDataSourceTableType = Filter(MyDataSource, Value > 10);C():MyDataSourceTableType = A(); B():MyDataSourceTableType = Filter(C(), Value > 11);", CultureInfo.InvariantCulture, symbolTable: recalcEngine.EngineSymbols, allowSideEffects: true);
             var func = recalcEngine.Functions.WithName("A");
 
-            if (func is UserDefinedFunction udf)
+            if (func.First() is UserDefinedFunction udf)

```

Comment:
>func.First() is UserDefinedFunction udf [](http://example.com/codeflow?start=16&length=39)

Instead of checking whether it is a UDF, consider casting it directly to UserDefinedFunction - so if it is not a UDF (which would be a bug) then the test would fail.